### PR TITLE
[Text Input] Only unfocus when receive connectionClosed on platform except web

### PR DESCRIPTION
**Pull request by luckysmg**
One of  https://github.com/flutter/flutter/issues/123523

Currently from engine side, in web engine we call `connectionClosed` from 
https://github.com/flutter/engine/blob/b789b197c2188717dd48a15c2188cb1a1300d397/lib/web_ui/lib/src/engine/text_editing/text_editing.dart#L2162

But no call from other platform. When on platform like iOS or android, we actually expect a simple unfocus operation without calling other things. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 1]
```diff
@@ -1429,6 +1429,57 @@ void main() {
...
+    editableText.connectionClosed();
+    await tester.pump();
+
+    if (kIsWeb) {
+      expect(onSubmittedInvoked, isTrue);
+      expect(onEditingCompleteInvoked, isTrue);
+      // Because we add the onEditingComplete and we didn't unfocus there, so focus still exists.
+      expect(focusNode.hasFocus, isTrue);
+    } else {
+      // For mobile and other platform, call connectionClosed will only unfocus.
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:1476]:
```
// For mobile and other platforms, calling connectionClosed will only unfocus.
```

**Review by justinmc — APPROVED** [review]
Are you sure that nothing in _finalizeEditing needs to be called when the connection is closed?  There could be people that are relying on that behavior.  It's reassuring that no tests have failed, though.  At least no one in the past has had a strong enough opinion about this to add a test for it.

**Comment by luckysmg** [comment]
Emm seems google test failed again😳😳

**Comment by goderbauer** [comment]
(triage) On first look, the google testing failures look unrelated. Can you rebase this PR with the latest master to trigger another run?

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 1 above)
Comment (luckysmg) [reply packages/flutter/test/widgets/editable_text_test.dart]: Done ^_^

**Comment by luckysmg** [comment]
> Are you sure that nothing in _finalizeEditing needs to be called when the connection is closed

@justinmc 
As for other platform except web, no call from engine. This will be called when platform close the connection first.
If we call `_finalizeEditing`, if `onEditComplete` is not null, it will execute so the text field will not focus, and if `onSubmit ` passed, it will be called too. But when platform close first, we just need a unfocus without submit anything. So I think this makes sense.

**Comment by luckysmg** [comment]
@goderbauer google test still failed😳😳

**Comment by goderbauer** [comment]
The Google Testing service is having some issues right now. We'll have to wait until those are fixed. Sorry about that.

**Comment by luckysmg** [comment]
All tests passed. Applying label for merging. ^_^

**Pull request merged**